### ISSUE TITLE
3.0.0-M2 Improved @ConstructorBinding Detection

### ISSUE THREAD

**Issue by fabapp2** [spring-projects-experimental/spring-boot-migrator#166]
### What needs to be done
Implement the required migration 

https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0.0-M2-Release-Notes#improved-constructorbinding-detection

Example code (to discuss)

#### TestCase 1
Given: the Spring Boot 2 has Configuration property file annotated with @ConstructorBinding and has only single constructor
Expected: the @ConstructorBinding annotation is removed for SpringBoot 3.0.X (it is not necessary anymore)

#### Acceptance criterias

**Report**
Given: class annotated with ...
When: recipe is applied..
Then: ....

**Recipe**
Given: class annotated with ...
When: recipe is applied..
Then: ....

Initial code:
```
@ConfigurationProperties(prefix = "mail")
@ConstructorBinding
public class ConfigProperties {
    private String hostName;
    private int port;
    private String from;

    public ConfigProperties(String hostName, int port, String from) {
        this.hostName = hostName;
        this.port = port;
        this.from = from;
    }


    public String getHostName() {
        return hostName;
    }

    public int getPort() {
        return port;
    }

    public String getFrom() {
        return from;
    }
}
```

Transformed code for SpringBoot 3.0.X:
```
@ConfigurationProperties(prefix = "mail")
public class ConfigProperties {
    private String hostName;
    private int port;
    private String from;

    public ConfigProperties(String hostName, int port, String from) {
        this.hostName = hostName;
        this.port = port;
        this.from = from;
    }


    public String getHostName() {
        return hostName;
    }

    public int getPort() {
        return port;
    }

    public String getFrom() {
        return from;
    }
}
```

**Comment by ashakirin** [comment]
TestCase 1: 
Given: the Spring Boot 2 has Configuration property file annotated with @ConstructorBinding and has only single constructor
Expected: the @ConstructorBinding annotation is removed for SpringBoot 3.0.X (it is not necessary anymore)

`@ConfigurationProperties(prefix = "mail")
@ConstructorBinding
public class ConfigProperties {
    private String hostName;
    private int port;
    private String from;

    public ConfigProperties(String hostName, int port, String from) {
        this.hostName = hostName;
        this.port = port;
        this.from = from;
    }


    public String getHostName() {
        return hostName;
    }

    public int getPort() {
        return port;
    }

    public String getFrom() {
        return from;
    }
}
`

**Comment by sanagaraj-pivotal** [comment]
![Screenshot 2022-06-22 at 11 43 25](https://user-images.githubusercontent.com/59915704/175010687-d8ccfc33-8bf0-40a7-aefe-5ffcc8a24d47.png)

**Comment by BoykoAlex** [comment]
I looked at this recipe already and contributed to `rewrite-spring` https://github.com/openrewrite/rewrite-spring/commit/5a37e8e8a846cf931d2e7966610b173566a85b62

I was looking at annotations over constructors as well but at the end decided to follow exactly what release note state without doing any further work:
@ConstructorBinding is no longer needed at the type level on @ConfigurationProperties classes and should be removed. When a class or record has multiple constructors, it may still be used on a constructor to indicate which one should be used for property binding.

Just removed the type annotation from the type annotated with `@ConfigurationProperties`. If `@ConstructorBinding` is present over type with multiple constructors - remove the annotation, likely the code wasn't working properly anyway. Also, single constructor annotated i was removing it at first but the release notes don't state that we should remove anything from constructors hence i left it in. However, if you feel we should make it better for the constructors as well - I don't mind

Since I've contributed it to rewrite-spring already perhaps further corrections (if needed) we can make in **rewrite-spring** directly?
